lib/go/edgecontext/edgecontext_test.go

```diff
@@ -240,6 +242,73 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestLocale(t *testing.T) {
+	for _, c := range []struct {
+		label  string
+		locale string
+		valid  bool
+	}{
+		{
+			label:  "valid-language",
+			locale: "es",
+			valid:  true,
+		},
+		{
+			label:  "valid-language-valid-region",
+			locale: "es_MX",
+			valid:  true,
+		},
+		{
+			label:  "valid-language-invalid-region",
+			locale: "es_MEX",
+			valid:  false,
+		},
+		{
+			label:  "invalid-language-valid-region",
+			locale: "esp_MX",
+			valid:  false,
+		},
+		{
+			label:  "invalid-language-invalid-region",
+			locale: "esp_MEX",
+			valid:  false,
+		},
+		{
+			label:  "invalid-separator",
+			locale: "es-MX",
+			valid:  false,
+		},
+		{
+			label:  "invalid-capitalization",
+			locale: "ES_MX",
+			valid:  false,
+		},
+	} {
+		t.Run(c.label, func(t *testing.T) {
+			err := edgeContextWithLocaleCode(c.locale)
+			if (err == nil) != c.valid {
+				if err == edgecontext.ErrInvalidLocaleCode {
```

Comment:
wait this is not right.

this is after `(err == nil) != c.valid`, which means this is only executed when `c.valid` is true, so this if block would never be hit.

what we want here is instead:
```go
if c.valid {
  if err != nil {
    t.Errorf("Did not expect error, got %v", err)
  }
} else {
  if !errors.Is(err, edgecontext.ErrInvalidLocaleCode) {
    t.Errorf("Expected edgecontext.ErrInvalidLocaleCode, got %v", err)
  }
}
```

Reply:
Either way this version is easier to reason about. Updated.